File: latchkey/browser_state.py
"""Browser state management utilities."""

import os
from pathlib import Path

LATCHKEY_BROWSER_STATE_ENV_VAR = "LATCHKEY_BROWSER_STATE"


def get_browser_state_path() -> Path | None:
    """Get the browser state path from the LATCHKEY_BROWSER_STATE environment variable."""
    env_value = os.environ.get(LATCHKEY_BROWSER_STATE_ENV_VAR)
    if env_value:
        return Path(env_value).expanduser()
    return None

File: port/src/browserState.ts
/**
 * Browser state management utilities.
 */

import { homedir } from "node:os";
import { resolve } from "node:path";

const LATCHKEY_BROWSER_STATE_ENV_VAR = "LATCHKEY_BROWSER_STATE";

/**
 * Get the browser state path from the LATCHKEY_BROWSER_STATE environment variable.
 */
export function getBrowserStatePath(): string | null {
  const envValue = process.env[LATCHKEY_BROWSER_STATE_ENV_VAR];
  if (envValue) {
    // Expand ~ to home directory
    if (envValue.startsWith("~")) {
      return resolve(homedir(), envValue.slice(2));
    }
    return resolve(envValue);
  }
  return null;
}
